src/os/bluestore/BlueStore.cc

```diff
@@ -9874,7 +9874,20 @@ void BlueStore::_kv_sync_thread()
 	if (!bluefs_extents_reclaiming.empty()) {
 	  dout(0) << __func__ << " releasing old bluefs 0x" << std::hex
 		   << bluefs_extents_reclaiming << std::dec << dendl;
+	  int r = 0;
+	  if (cct->_conf->bdev_enable_discard && cct->_conf->bdev_async_discard) {
+	    r = bdev->queue_discard(bluefs_extents_reclaiming);
+	    if (r == 0) {
+	      goto clear;
+	    }
+	  } else if (cct->_conf->bdev_enable_discard) {
+	    for (auto p = bluefs_extents_reclaiming.begin(); p != bluefs_extents_reclaiming.end(); ++p) {
+	      bdev->discard(p.get_start(), p.get_len());
+	    }
+	  }
+
 	  alloc->release(bluefs_extents_reclaiming);
+clear:
```

Comment:
Could you please fix the same code at _txc_release_alloc() to avoid double release too? Using separate commit please.

And may be refactor this patch to avoid copy-paste.

Reply:
I did not find any double release at _txc_release_alloc(), could you please point it out for me?

Reply:
Nevermind. Sorry, I missed "goto out" there.